Formula/dtrx.rb

```diff
@@ -0,0 +1,25 @@
+class Dtrx < Formula
+  include Language::Python::Virtualenv
+
+  desc "Intelligent archive extraction"
+  homepage "https://github.com/dtrx-py/dtrx/"
+  url "https://files.pythonhosted.org/packages/25/cb/1ef093d762f4d5963e9e571daec239acc5f4971eb9daeda77b131d7cf39f/dtrx-8.3.1.tar.gz"
+  sha256 "5587258e762074d5395a6824fd7968ca4f4a1dc481f4852fb84d14e7624433fb"
+  license "GPL-3.0-only"
+
+  # Include a few common decompression handlers in addition to the python dep
+  depends_on "p7zip"
+  depends_on "python@3.9"
+  depends_on "xz"
+  depends_on "zip"
+  uses_from_macos "bzip2"
+  uses_from_macos "unzip"
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "#{bin}/dtrx", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
:facepalm: apologies, sorry for missing that part of the manual! added a simple test.